qiling/loader/pe.py

```diff
@@ -349,17 +349,17 @@ def __init__(self, ql):
         self.path       = self.ql.path
 
     def run(self):
-        self.init_dlls = [b"ntoskrnl.exe", b"ntdll.dll", b"kernel32.dll", b"user32.dll"]
-        self.sys_dlls = [b"ntoskrnl.exe", b"ntdll.dll", b"kernel32.dll"]
+        self.init_dlls = [b"ntdll.dll", b"kernel32.dll", b"user32.dll"]
+        self.sys_dlls = [b"ntdll.dll", b"kernel32.dll"]
         self.pe_entry_point = 0
         self.sizeOfStackReserve = 0        
 
-        if self.ql.shellcoder:
-            self.init_dlls.remove(b"ntoskrnl.exe")
-            self.sys_dlls.remove(b"ntoskrnl.exe")
-        else:
+        if not self.ql.shellcoder:
             self.pe = pefile.PE(self.path, fast_load=True)
             self.is_driver = (self.pe.OPTIONAL_HEADER.Subsystem == 1)
+            if self.is_driver:
+                self.init_dlls = [b"ntoskrnl.exe"]
+                self.sys_dlls = [b"ntoskrnl.exe"]
```

Comment:
Nothing should import from both ntoskrnl.exe and ntdll.dll
Considering the 'shellcoder' case already removed ntoskrnl, this should leave that case in the same state.
The is_driver case now completely drops those usermode dll's, and replaces them all with ntoskrnl.
Services (f.e. svchost.exe or umpnpmgr.dll) Do not have the Subsystem `1` set, and import from ntdll and more. (and not ntoskrnl)